include/util/enum.hpp

```diff
@@ -12,12 +12,26 @@ template <typename EnumType>
 struct EnumParser {
   EnumParser() {}
 
+  // Helper function to capitalize a string
```

Comment:
Two more things:

* I would put this function in a separate file "utils/strings.hpp"
* It's better to declare the functions in the hpp, but implement them in a cpp file, otherwise compilation might take longer: https://stackoverflow.com/a/1809753/10180569. It's not really a problem right now (because we're only including it in one place), but it's good practice to change it.

Reply:
That makes me kinda want to go through and clean up all the other poor separations if it can affect that much during compilation.